data.sql

```diff
@@ -94,3 +94,25 @@ INSERT INTO visits (animal_id, vet_id, date_of_visit)
 VALUES ((SELECT id FROM animals WHERE name = 'Blossom'), (SELECT id FROM vets WHERE name = 'Stephanie Mendez'), '2020-05-24');
 INSERT INTO visits (animal_id, vet_id, date_of_visit)
 VALUES ((SELECT id FROM animals WHERE name = 'Blossom'), (SELECT id FROM vets WHERE name = 'William Tatcher'), '2021-01-11');
+
+-- This will add 3.594.280 visits considering you have 10 animals, 4 vets, and it will use around ~87.000 timestamps (~4min approx.)
+INSERT INTO visits (animal_id, vet_id, date_of_visit) SELECT * FROM (SELECT id FROM animals) animal_ids, (SELECT id FROM vets) vets_ids, generate_series('1980-01-01'::timestamp, '2021-01-01', '4 hours') visit_timestamp;
+
+-- This will add 2.500.000 owners with full_name = 'Owner <X>' and email = 'owner_<X>@email.com' (~2min approx.)
+insert into owners (full_name, email) select 'Owner ' || generate_series(1,2500000), 'owner_' || generate_series(1,2500000) || '@mail.com';
+
+
+/* Decreasing execution times */
+
+EXPLAIN ANALYZE SELECT COUNT(*) FROM visits WHERE animal_id=4;
+CREATE INDEX idx_visits_animal_id ON visits (animal_id);
+EXPLAIN ANALYZE SELECT COUNT(*) FROM visits where animal_id = 4;
+
+
+EXPLAIN ANALYZE SELECT * FROM visits where vet_id = 2;
+CREATE INDEX idx_visits_vet_id ON visits (vet_id);
+EXPLAIN ANALYZE SELECT * FROM visits where vet_id = 2;
+
+EXPLAIN ANALYZE SELECT * FROM owners where email = 'owner_18327@mail.com';
+CREATE INDEX idx_owners_email ON owners (email);
```

Comment:
Good job resolving the n + 1 problem with the database, however, the commands that create an index on the table should be in the schema.sql file. Consider doing so to correctly separate db set up from db operation

Reply:
@DuaneDave Thanks! one question, just the create index of the three of the right? not with the explain analyze?